adapters/onebot/src/bot/message.ts

```diff
@@ -1,9 +1,10 @@
 import { Messenger, pick, segment, Universal } from '@satorijs/satori'
 import { BaseBot } from './base'
 import { CQCode } from './cqcode'
+import { Author } from '../types'
 
 class State {
-  author: Partial<Universal.Author> = {}
+  author: Partial<Author> | string = {}
```

Comment:
I reread the mechanic about this PR.
I think you could just make `author` as `Author`, which has a property `messageId`, assign the `author.messageId = attr.id.toString()`. then just `if (state.messageId) ... else ...`

You even no need to check the type of the `author` variable.

Reply:
lgtm